.github/workflows/node-ci.yml

```diff
@@ -0,0 +1,181 @@
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node-version: [16.x]
+    steps:
+    - name: Checking out latest commit
+      uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Cache node modules
+      id: node-modules-cache # use this to check for `cache-hit` (`steps.node-modules-cache.outputs.cache-hit != 'true'`)
+      uses: actions/cache@v2
+      with:
+        path: node_modules
+        key: ${{ runner.os }}-${{ matrix.node-version }}-node-modules-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.node-version }}-node-modules-
+    - name: Install dependencies
+      if: steps.node-modules-cache.outputs.cache-hit != 'true'
+      run: yarn install --frozen-lockfile
+
+    - name: Get All Changed Files
+      uses: jaredly/get-changed-files@v1.0.1
+      id: changed
+      with:
+        format: space-delimited
+        absolute: true
+
+    - name: Build so that inter-package references are resolved
+      run: yarn build
+
+    # Linting / type checking
+    - name: Run ESLint (all files)
+      if: contains(steps.changed.outputs.added_modified, '.eslintrc.js') || contains(steps.changed.outputs.added_modified, 'yarn.lock')
+      run: |
+        yarn lint:ci packages
+    - name: Run ESLint (added/changed files)
+      if: contains(steps.changed.outputs.added_modified, '.eslintrc.js') == false && contains(steps.changed.outputs.added_modified, 'yarn.lock') == false
+      run: |
+        yarn lint:ci ${{ steps.changed.outputs.added_modified }}
+    - name: Run Flow
+      uses: Khan/flow-action@v1.1.3
+      with:
+        flow-bin: ./node_modules/.bin/flow
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    # Collect and record coverage
+    - name: Run Jest with coverage
+      run: yarn coverage
+    # - name: Upload Coverage
+    #   # We don't bother collecting a record of coverage for dependabot changes
+    #   if: ${{ github.actor != 'dependabot[bot]' }}
+    #   uses: codecov/codecov-action@v2
+    #   with:
+    #     token: ${{ secrets.CODECOV_TOKEN }}
+    #     files: ./coverage/coverage-final.json
+    #     fail_ci_if_error: true
+
+    # Make sure our packages aren't growing unexpectedly
+    # This must come last as it builds the old code last and so leaves the
+    # wrong code in place for the next job.
+    # - name: Check Builds
+    #   uses: preactjs/compressed-size-action@v2
+    #   with:
+    #     # We only care about the browser ES module size, really:
+    #     pattern: "**/dist/es/index.browser.js"
+    #     # Always ignore SourceMaps and node_modules:
+    #     exclude: "{**/*.map,**/node_modules/**}"
+    #     # Clean up before a build
+    #     clean-script: "clean"
+    #     # Build production
+    #     build-script: "build:prodsizecheck"
```

Comment:
I'll re-enable this after this PR lands.  This step gets confused because this is the initial commit.